salt/metalk8s/volumes/prepared/init.sls

```diff
@@ -0,0 +1,51 @@
+include:
+  - .installed
+
+Refresh pillars for all minions:
+  salt.function:
+    - name: saltutil.refresh_pillar
+    - tgt: '*'
+
+{%- set volumes_to_create = [] %}
+{%- set all_volumes = pillar.metalk8s.volumes %}
+{%- set target_volume = pillar.get('volume') %}
+
+{#- If a volume is given we only take this one, otherwise we take them all. #}
+{%- if target_volume is not none %}
+  {%- if target_volume in all_volumes.keys() %}
+    {%- do volumes_to_create.append(target_volume) %}
+  {%- else %}
+Volume "{{ target_volume }}" not found in pillar:
+  test.fail_without_changes: []
+  {%- endif %}
+{%- else %}
+    {%- do volumes_to_create.extend(all_volumes.keys()|list) %}
+{%- endif %}
+
+Create the sparse file spool:
+  file.directory:
+    - name: /var/lib/metalk8s/storage/sparse/
+    - makedirs: True
+
+{%- for volume in volumes_to_create %}
+
+Create backing storage for {{ volume }}:
+  metalk8s_volumes.present:
+    - volume: {{ volume }}
+    - require:
+      - file: Create the sparse file spool
```

Comment:
Strictly speaking this is only required for `sparseLoopDevice` volumes, but fine.

Reply:
Yeah, it avoids the need for an `if` in the Jinja and it simplifies the looping when you create all the volumes.

We could also update the module to create the parent directories when creating the sparse file, but that's less explicit.